### PR TITLE
✨ source-paypal: make the time window configurable

### PR DESCRIPTION
**Pull request by girarda**
## What
* Allow users to configure the size of the time window each requested by each HTTP request

## How
* Add an optional field to the spec defining the number of days each HTTP request should fetch
* Update the `DatetimeBasedCursor`'s `step` to be configured through the config object
* Update the docs so users know they should lower the size of the time window if they hit an error saying the result set size is too large

## Recommended reading order
1. `airbyte-integrations/connectors/source-paypal-transaction/source_paypal_transaction/manifest.yaml`
2. `docs/integrations/sources/paypal-transaction.md`
3. `airbyte-integrations/connectors/source-paypal-transaction/metadata.yaml`

## 🚨 User Impact 🚨
Users of source-paypal-transaction can optionally configure the size of the time window

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- Connector version is set to `0.0.1`
    - `Dockerfile` has version `0.0.1`
- Documentation updated
    - Connector's `README.md`
    - Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - `docs/integrations/README.md`

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Connector Generator</strong></summary>

- Issue acceptance criteria met
- PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/resources/pull-requests-handbook)
- If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:generateScaffolds` then checking in your changes
- Documentation which references the generator is updated as needed

</details>

<details><summary><strong>Updating the Python CDK</strong></summary>

### Airbyter

Before merging:
- Pull Request description explains what problem it is solving
- Code change is unit tested
- Build and my-py check pass
- Smoke test the change on at least one affected connector
   - On Github: Run [this workflow](https://github.com/airbytehq/airbyte/actions/workflows/connectors_tests.yml), passing `--use-local-cdk --name=source-<connector>` as options
   - Locally: `airbyte-ci connectors --use-local-cdk --name=source-<connector> test`
- PR is reviewed and approved
      
After merging:
- [Publish the CDK](https://github.com/airbytehq/airbyte/actions/workflows/publish-cdk-command-manually.yml)
   - The CDK does not follow proper semantic versioning. Choose minor if this the change has significant user impact or is a breaking change. Choose patch otherwise.
   - Write a thoughtful changelog message so we know what was updated.
- Merge the platform PR that was auto-created for updating the Connector Builder's CDK version
   - This step is optional if the change does not affect the connector builder or declarative connectors.

</details>

### CODE REVIEW

**Comment by airbyte-oss-build-runner** [comment]
##  source-paypal-transaction test report (commit [`67b779ed68`](https://github.com/airbytehq/airbyte/pull/31852/commits/67b779ed689cea282ab26b97e4cbe4476c574d90)) - ❌

⏲️  Total pipeline duration: 33.89s 

| Step                                                                     | Result   |
|:-------------------------------------------------------------------------|:---------|
| Build source-paypal-transaction docker image for platform(s) linux/amd64 | ✅       |
| Acceptance tests                                                         | ❌       |
| Check our base image is used                                             | ✅       |
| Code format checks                                                       | ✅       |
| Validate metadata for source-paypal-transaction                          | ✅       |
| Connector version semver check                                           | ✅       |
| Connector version increment check                                        | ✅       |
| QA checks                                                                | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/alex_paypal_configurable_time_window/1698275762/67b779ed689cea282ab26b97e4cbe4476c574d90/source-paypal-transaction/2.1.3/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:67b779ed689cea282ab26b97e4cbe4476c574d90)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-paypal-transaction test
```

File: airbyte-integrations/connectors/source-paypal-transaction/metadata.yaml [hunk 1]
```diff
@@ -11,7 +11,7 @@ data:
   connectorSubtype: api
   connectorType: source
   definitionId: d913b0f2-cc51-4e55-a44c-8ba1697b9239
-  dockerImageTag: 2.1.2
+  dockerImageTag: 2.1.3
```
Comment (erohmensing) [inline airbyte-integrations/connectors/source-paypal-transaction/metadata.yaml:14]: Nit: I would say this is a minor as it's new functionality

**Review by erohmensing — APPROVED** [review]
Nice! And the test you ran for the user failing on 7D was with 1D?

File: airbyte-integrations/connectors/source-paypal-transaction/metadata.yaml (hunk 1 above)
Comment (girarda) [reply airbyte-integrations/connectors/source-paypal-transaction/metadata.yaml]: good call. fixed!

**Comment by girarda** [comment]
@erohmensing 
> Nice! And the test you ran for the user failing on 7D was with 1D?

Yes, I tested and succeeded with 1D. There might be some value in figuring out the exact window that will succeed for them, but a full refresh only took 45 minutes with 1D so I don't think it's worth the trouble right now

**Comment by erohmensing** [comment]
@girarda sounds great!

**Comment by airbyte-oss-build-runner** [comment]
##  source-paypal-transaction test report (commit [`b7978c4c3e`](https://github.com/airbytehq/airbyte/pull/31852/commits/b7978c4c3e9f35434ad901cf2fce0c413abe664c)) - ❌

⏲️  Total pipeline duration: 29mn08s 

| Step                                                                     | Result   |
|:-------------------------------------------------------------------------|:---------|
| Build source-paypal-transaction docker image for platform(s) linux/amd64 | ✅       |
| Acceptance tests                                                         | ❌       |
| Check our base image is used                                             | ✅       |
| Code format checks                                                       | ✅       |
| Validate metadata for source-paypal-transaction                          | ✅       |
| Connector version semver check                                           | ✅       |
| Connector version increment check                                        | ✅       |
| QA checks                                                                | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/alex_paypal_configurable_time_window/1698277687/b7978c4c3e9f35434ad901cf2fce0c413abe664c/source-paypal-transaction/2.2.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:b7978c4c3e9f35434ad901cf2fce0c413abe664c)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-paypal-transaction test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-qualaroo test report (commit [`d7715568ba`](https://github.com/airbytehq/airbyte/pull/31852/commits/d7715568baedffd4b3148579c43e0dff644055a4)) - ✅

⏲️  Total pipeline duration: 01mn14s 

| Step                                                           | Result   |
|:---------------------------------------------------------------|:---------|
| Build source-qualaroo docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                                     | ✅       |
| Acceptance tests                                               | ✅       |
| Code format checks                                             | ✅       |
| Validate metadata for source-qualaroo                          | ✅       |
| Connector version semver check                                 | ✅       |
| QA checks                                                      | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/alex_paypal_configurable_time_window/1698293883/d7715568baedffd4b3148579c43e0dff644055a4/source-qualaroo/0.3.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:d7715568baedffd4b3148579c43e0dff644055a4)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-qualaroo test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-qualaroo test report (commit [`910a92e063`](https://github.com/airbytehq/airbyte/pull/31852/commits/910a92e06322c8831769af1b08ea82bf81b5a609)) - ✅

⏲️  Total pipeline duration: 34.06s 

| Step                                                           | Result   |
|:---------------------------------------------------------------|:---------|
| Build source-qualaroo docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                                     | ✅       |
| Acceptance tests                                               | ✅       |
| Code format checks                                             | ✅       |
| Validate metadata for source-qualaroo                          | ✅       |
| Connector version semver check                                 | ✅       |
| QA checks                                                      | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/alex_paypal_configurable_time_window/1698294835/910a92e06322c8831769af1b08ea82bf81b5a609/source-qualaroo/0.3.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:910a92e06322c8831769af1b08ea82bf81b5a609)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-qualaroo test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-paypal-transaction test report (commit [`910a92e063`](https://github.com/airbytehq/airbyte/pull/31852/commits/910a92e06322c8831769af1b08ea82bf81b5a609)) - ✅

⏲️  Total pipeline duration: 02mn25s 

| Step                                                                     | Result   |
|:-------------------------------------------------------------------------|:---------|
| Build source-paypal-transaction docker image for platform(s) linux/amd64 | ✅       |
| Acceptance tests                                                         | ✅       |
| Check our base image is used                                             | ✅       |
| Code format checks                                                       | ✅       |
| Validate metadata for source-paypal-transaction                          | ✅       |
| Connector version semver check                                           | ✅       |
| Connector version increment check                                        | ✅       |
| QA checks                                                                | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/alex_paypal_configurable_time_window/1698294835/910a92e06322c8831769af1b08ea82bf81b5a609/source-paypal-transaction/2.2.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:910a92e06322c8831769af1b08ea82bf81b5a609)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-paypal-transaction test
```

**Pull request merged**